### PR TITLE
feat(FX-4538): Hide AP blue dot indicator if there are no new notifications

### PR DESCRIPTION
**Pull request by nickskalkin**
Co-authored-by: Dima Tretyak <itretyak@ya.ru>

The type of this PR is: **Feat**

This PR solves [FX-4538]

### Description

As a user
Given I have unseen notifications
When I visit the activity panel
The top-level indicator goes away
And I continue to see unread indicators to individual notifications.

### Demo

- [Review app](https://notifications-bell.artsy.net/) - will work only after https://github.com/artsy/metaphysics/pull/4731 is merged

Might be convenient to use this mutation to imitative opening notifications feed in past:

```graphql
mutation {
	markNotificationsAsSeen(input: { until: "2023-01-28T11:05:19Z"}) {
		responseOrError {
			... on MarkNotificationsAsSeenSuccess {
				success
			}
			
			... on MarkNotificationsAsSeenFailure {
				mutationError {
					message
				}
			}
		}
	}
}
```

#### Before

https://user-images.githubusercontent.com/3934579/215535046-59477027-0543-4ca3-af60-2172089c2f08.mp4

#### After

https://user-images.githubusercontent.com/3934579/215535089-b831a64d-e8e2-4ac3-bf7b-370f3b08bcf2.mp4

[FX-4538]: https://artsyproduct.atlassian.net/browse/FX-4538?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Notifications/Notifications.tsx [hunk 1]
```diff
@@ -11,6 +18,32 @@ export const Notifications: React.FC<NotificationsProps> = ({
   paginationType,
   ...rest
 }) => {
+  const { relayEnvironment } = useSystemContext()
+
+  const markAsSeen = useCallback(async () => {
```
Comment (dimatretyak) [inline src/Components/Notifications/Notifications.tsx:23]: we can remove `useCallback` here

File: src/Components/Notifications/Notifications.tsx [hunk 2]
```diff
@@ -2,6 +2,13 @@ import { Tab } from "@artsy/palette"
 import { NofiticationsTabs, NofiticationsTabsProps } from "./NotificationsTabs"
 import { NotificationsListQueryRenderer } from "./NotificationsList"
 import { NotificationPaginationType } from "./types"
+import { useEffect, useCallback } from "react"
+import { DateTime } from "luxon"
+import { markNotificationsAsSeen } from "./Mutations/markNotificationsAsSeen"
+import { useSystemContext } from "System"
+import createLogger from "Utils/logger"
+
+const logger = createLogger("MarkNotificationsAsSeen")
```
Comment (dimatretyak) [inline src/Components/Notifications/Notifications.tsx:11]: usually we pass the component name to `createLogger`. Something like this`const logger = createLogger("Notifications")`

File: src/Components/NavBar/NavBarMobileMenu/NavBarMobileMenuNotifications.tsx [hunk 3]
```diff
@@ -25,8 +25,11 @@ export const NavBarMobileMenuNotifications: React.FC<NavBarMobileMenuNotificatio
   const { trackEvent } = useTracking()
   const unreadConversationCount = me?.unreadConversationCount ?? 0
   const unreadNotificationsCount = me?.unreadNotificationsCount ?? 0
+  const unseenNotificationsCount = me?.unseenNotificationsCount ?? 0
   const hasConversations = unreadConversationCount > 0
   const hasNotifications = unreadNotificationsCount > 0
+  const hasUnseenNotifications = unseenNotificationsCount > 0
+  const displayBlueDot = hasNotifications && hasUnseenNotifications
```
Comment (dimatretyak) [inline src/Components/NavBar/NavBarMobileMenu/NavBarMobileMenuNotifications.tsx:32]: **suggestion**: rename `displayBlueDot` to `shouldDisplayIndicator`. what do you think?

File: src/Components/Notifications/Mutations/markNotificationsAsSeen.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+const updater = (
+  store: RecordSourceSelectorProxy<markNotificationsAsSeenMutation$data>
+) => {
+  const root = store.getRoot()
+  const me = root.getLinkedRecord("me")
+
+  if (!me) {
+    return
+  }
+  // Set unread notifications count to 0
```
Comment (dimatretyak) [inline src/Components/Notifications/Mutations/markNotificationsAsSeen.ts:21]: **chore**: `Set unseen notifications count to 0`

File: src/Components/NavBar/NavBarMobileMenu/NavBarMobileMenuNotifications.tsx (hunk 3 above)
Comment (nickskalkin) [reply src/Components/NavBar/NavBarMobileMenu/NavBarMobileMenuNotifications.tsx]: done!

Comment (dimatretyak) [reply src/Components/NavBar/NavBarMobileMenu/NavBarMobileMenuNotifications.tsx]: thanks!

**Review by dimatretyak — APPROVED** [review]
LGTM 👍

**Review by gkartalis — APPROVED** [review]
⭐

**Pull request merged**